pybop/problems/fitting_problem.py

```diff
@@ -134,3 +135,112 @@ def evaluateS1(self, inputs: Inputs):
         )
 
         return (y, np.asarray(dy))
+
+
+class MultiFittingProblem(BaseProblem):
+    """
+    Problem class for joining mulitple fitting problems.
+
+    Extends `FittingProblem` to multiple fitting problems.
+    """
+
+    def __init__(self, problem_list, weights=None):
+        self.problem_list = problem_list
+        self.weights = weights
+
+        # Compile the set of parameters, ignoring duplicates
+        combined_parameters = Parameters()
+        for problem in self.problem_list:
+            combined_parameters.join(problem.parameters)
+
+        # Combine the target datasets
+        combined_dataset = Dataset(
+            {"Time [s]": np.asarray([]), "Combined signal": np.asarray([])}
+        )
+        for problem in self.problem_list:
+            for signal in problem.signal:
+                combined_dataset["Time [s]"] = np.concatenate(
+                    (combined_dataset["Time [s]"], problem._time_data)
+                )
+                combined_dataset["Combined signal"] = np.concatenate(
+                    (combined_dataset["Combined signal"], problem._target[signal])
+                )
+
+        super().__init__(
+            parameters=combined_parameters,
+            model=None,
+            signal=["Combined signal"],
+        )
+        self._dataset = combined_dataset.data
+        self.parameters.initial_value()
+
+        # Unpack time and target data
+        self._time_data = self._dataset["Time [s]"]
+        self.n_time_data = len(self._time_data)
+        self.set_target(combined_dataset)
+
+    def evaluate(self, inputs: Inputs):
+        """
+        Evaluate the model with the given parameters and return the signal.
+
+        Parameters
+        ----------
+        inputs : Inputs
+            Parameters for evaluation of the model.
+
+        Returns
+        -------
+        y : np.ndarray
+            The model output y(t) simulated with given inputs.
+        """
+        inputs = self.parameters.verify(inputs)
+        self.parameters.update(values=list(inputs.values()))
+
+        y = {"Combined signal": np.asarray([])}
+        for problem in self.problem_list:
+            problem_inputs = problem.parameters.as_dict()
+            for signal in problem.signal:
+                yi = problem.evaluate(problem_inputs)
+                y["Combined signal"] = np.concatenate(
+                    (y["Combined signal"], yi[signal])
+                )
+
+        return y
```

Comment:
Same as the `evaluateS1()` below.

```suggestion
        combined_signal = []
        
        for problem in self.problem_list:
            problem_inputs = problem.parameters.as_dict()
            signal_values = problem.evaluate(problem_inputs)
            
            # Collect signals
            for signal in problem.signal:
                combined_signal.extend(signal_values[signal])
        
        return {"Combined signal": np.asarray(combined_signal)}
```